src/components/SQFormDialogStepper/SQFormDialogStepper.tsx

```diff
@@ -12,66 +11,119 @@ import {
+  children?: JSX.Element;
+  /** Should the loading spinner be shown */
+  isLoading?: boolean;
+  /** Optional message to be added to the loading spinner */
+  loadingMessage?: string;
+}
 
 export function SQFormDialogStep({
   children,
   isLoading = false,
-  loadingMessage = ''
-}) {
+  loadingMessage = '',
+}: SQFormDialogStepProps): JSX.Element {
   return isLoading ? (
     <LoadingSpinner message={loadingMessage} />
   ) : (
     <>{children}</>
   );
 }
 
-const Transition = React.forwardRef((props, ref) => {
+const Transition = React.forwardRef(function Transition(
+  props: TransitionProps & {children?: React.ReactElement<unknown, string>},
+  ref: React.Ref<unknown>
+) {
   return <Slide direction="down" ref={ref} {...props} />;
 });
 
 const useStyles = makeStyles({
   root: {
     padding: 20,
     width: '100%',
-    maxWidth: ({steps}) => `${300 * steps.length}px`,
+    maxWidth: ({
+      steps,
+    }: {
+      steps: Array<React.ReactChild | React.ReactFragment | React.ReactPortal>;
+    }) => `${300 * steps.length}px`,
     '& svg': {
       fontSize: 30,
       '& text': {
         fontSize: 15,
-        fontWeight: 600
-      }
+        fontWeight: 600,
+      },
     },
     '& span': {
-      whiteSpace: 'nowrap'
+      whiteSpace: 'nowrap',
     },
     '& .MuiStepLabel-label.MuiStepLabel-alternativeLabel': {
-      marginTop: '5px'
-    }
-  }
+      marginTop: '5px',
+    },
+  },
 });
 
 const useActionsStyles = makeStyles({
   root: {
     display: 'flex',
     justifyContent: 'space-between',
     flex: '1 1 100%',
-    padding: '16px 24px'
-  }
+    padding: '16px 24px',
+  },
 });
 
 const useStepperStyles = makeStyles({
   root: {
     padding: '1px',
-    justifyContent: 'center'
-  }
+    justifyContent: 'center',
+  },
 });
 
+interface SQFormDialogStepperProps {
+  /** The secondary button text (Button located on left side of Dialog) */
+  cancelButtonText?: string;
+  /** The content to be rendered in the dialog body.  Will be an array of React elements. */
+  children: Array<JSX.Element>;
+  /** If true, clicking the backdrop will not fire the onClose callback. */
+  disableBackdropClick?: boolean;
+  /** Sets the dialog to the maxWidth. */
+  fullWidth?: boolean;
+  /** Disables the next/submit button */
+  isNextDisabled?: boolean;
+  /** The current open/closed state of the Dialog */
+  isOpen: boolean;
+  /** Allows the initial values to be updated after initial render */
+  enableReinitialize?: boolean;
+  /** Determine the max-width of the dialog. The dialog width grows with the size of the screen. Set to false to disable maxWidth. */
+  maxWidth?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | false;
+  /** Callback function invoked when the user clicks on the secondary button or outside the Dialog */
+  onClose: () => void;
+  /** Callback function invoked when user clicks primary submit button */
+  onSubmit: (values: Record<string, unknown>, helpers: unknown) => void;
```

Comment:
I think we want to use generics for this component like we did with SQFormDialog. So that the form values will equal exactly what they're supposed to.

I do think we might need to get a little furth and have the generic `extend Record<string, unknown>` so we can index it properly in our comopnents maybe? That may not be necessary, however.

using a generic would affect the `onSubmit` function, `initialValues`, and looks like `setValues`